iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -2504,4 +2504,190 @@ TYPED_TEST(stringTyped_test, AccessMaxPositionOfNotEmptyStringViaConstSubscriptO
+}
+
+TEST(String10, InsertStringLiteralInTheMiddleOfTheStringSucceeds)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "861eec68-3e78-42cf-ac75-30c23a3e60d2");
+    string<10> sut("Hypoad");
+    ASSERT_TRUE(sut.insert(3, "not", 3));
+    EXPECT_THAT(sut.size(), Eq(9));
+    EXPECT_THAT(sut.c_str(), StrEq("Hypnotoad"));
+}
+
+TEST(String10, InsertStringLiteralAtTheEndOfTheStringSucceeds)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "91f733e0-5ded-4f85-9727-241b649bd7dc");
+    string<10> sut("Hypno");
+    ASSERT_TRUE(sut.insert(5, "toad", 4));
+    EXPECT_THAT(sut.size(), Eq(9));
+    EXPECT_THAT(sut.c_str(), StrEq("Hypnotoad"));
+}
+
+TYPED_TEST(stringTyped_test, InsertStringLiteralToEmptyStringWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "eba320fc-9909-4043-a944-7a5c9f2a4272");
+    ASSERT_TRUE(this->testSubject.insert(0, "M", 1));
+    EXPECT_THAT(this->testSubject.size(), Eq(1));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq("M"));
+}
+
+TYPED_TEST(stringTyped_test, InsertEmptyStringLiteralDoesNotChangeTheString)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "37d723b7-5b35-4f1d-ac33-23dd483121a1");
+    this->testSubject = "M";
+    ASSERT_TRUE(this->testSubject.insert(0, "", 0));
+    EXPECT_THAT(this->testSubject.size(), Eq(1));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq("M"));
+}
+
+TYPED_TEST(stringTyped_test, InsertStringLiteralWithCountGreaterThanSizeOfStringLiteralFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "3d2951f8-d868-4998-bb33-58424aba7a65");
+    ASSERT_FALSE(this->testSubject.insert(0, "M", 2));
+    EXPECT_THAT(this->testSubject.size(), Eq(0));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq(""));
+}
+
+TEST(String10, InsertTooLargeStringLiteralFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "0f11f387-f437-4a39-8207-259df4807c70");
+    string<10> sut("Ferdinand");
+    ASSERT_FALSE(sut.insert(9, "Spitzschnueffler", 16));
+    EXPECT_THAT(sut.size(), Eq(9));
+    EXPECT_THAT(sut.c_str(), StrEq("Ferdinand"));
+}
+
+TEST(String10, InsertTooLargeStringLiteralWithSmallCountSucceeds)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2763b3d8-3023-4841-80d1-798acd3b71d7");
+    string<10> sut("Ferdinand");
+    ASSERT_TRUE(sut.insert(9, "Spitzschnueffler", 1));
+    EXPECT_THAT(sut.size(), Eq(10));
+    EXPECT_THAT(sut.c_str(), StrEq("FerdinandS"));
+}
+
+TYPED_TEST(stringTyped_test, InsertStringLiteralWithCount0DoesntChangeTheString)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "90e80302-1c14-48bb-b9e9-0f8fca017fea");
+    ASSERT_TRUE(this->testSubject.insert(0, "Ferdinand", 0));
+    EXPECT_THAT(this->testSubject.size(), Eq(0));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq(""));
+}
+
+TEST(String10, InsertStringLiteralAtPositionGreaterStringSizeFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a3a30ea5-f228-48f4-9497-08502e4f2c9a");
+    string<10> sut("Muesli");
+    ASSERT_FALSE(sut.insert(7, "s", 1));
```

Comment:
Could you use a `pos` way larger than `7` here? My inner chicken always fears an of-by-one-error is these cases. Maybe this is yet another test case.

Reply:
Since I want to exclude an off-by-one error in the implementation, I chose a position of `sut.size() + 1` on purpose. I will make this clearer in the code.